coastlines/raster.py

```diff
@@ -49,6 +49,94 @@
 warnings.simplefilter(action="ignore", category=FutureWarning)
 
 
+
+def hillshade(dem, elevation, azimuth, vert_exag=1, dx=30, dy=30):
+
+    # For each supplied azimuth, compute hillshade
+    from matplotlib.colors import LightSource
+
+    hs = LightSource(azdeg=azimuth, altdeg=elevation).hillshade(
+        dem, vert_exag=vert_exag, dx=dx, dy=dy
+    )
+    return hs
+
+
+def terrain_shadow(x, elevation, threshold=0.4, radius=3):
+    hs = hillshade(elevation, x.sun_elevation, x.sun_azimuth)
+
+    from skimage.morphology import binary_dilation, binary_opening, disk
```

Comment:
I think you should do this import at the start of the function. So switch this and the `hs = ` lines